lib/SILGen/SILGenApply.cpp

```diff
@@ -612,9 +621,18 @@ class Callee {
         methodVal = SGF.emitClassMethodRef(
             Loc, borrowedSelf->getValue(), *constant, methodTy);
       } else {
-        methodVal = SGF.B.createObjCMethod(
-            Loc, borrowedSelf->getValue(), *constant,
-            SILType::getPrimitiveObjectType(methodTy));
+        if (constant->getAbstractFunctionDecl()->isObjC()) {
+          methodVal =
+              SGF.B.createObjCMethod(Loc, borrowedSelf->getValue(), *constant,
+                                     SILType::getPrimitiveObjectType(methodTy));
+        } else {
+          // TODO: Update borrowedSelf with the updated self-pointer...
```

Comment:
This seems important…

Reply:
When you do get to it, we'll need some way to indicate that the adjusted "this" pointer depends on the lifetime of the original object. I'm not sure if there's a way to do that other than the `mark_dependence` instruction, but there ought to be.

Reply:
My suggestion (going back to what I was saying above) is that the method should take self either forwarding or guaranteed.